middleware/index.ts

```diff
@@ -149,14 +161,37 @@ function parseCloudEvent(logger: Logger, headers: CEHeaders, body: any): CloudEv
         logger.info('Forced content-type to: application/cloudevents+json');
     }
 
+    const isSpec1 = parseInt(body.specversion?.split('.')?.[0] || 0) >= 1;
+    if (isSpec1 && body?.sfcontext) {
+      body['sfcontext'] = decode64(body.sfcontext);
+    } else if (body.data?.context) {
+      body['sfcontext'] = body.data.context;
+    }
+
+    if (isSpec1 && body?.sffncontext) {
+      body['sffncontext'] = decode64(body.sffncontext);
+    } else if (body.data?.sfContext) {
+      body['sffncontext'] = body.data.sfContext;
+    }
+
+    if (body?.data?.context && body?.data?.sfContext && body?.data?.payload) {
+      body.data = body.data?.payload;
+    }
+
     // make a clone of the body if object - cloudevents sdk deletes keys as it parses.
     // otherwise Receiver will do a JSON parse so need to re-stringify any string body
     const bodyShallowCopy = bodyIsObj ? Object.assign({}, body) :
             JSON.stringify(body);
     return Receiver.accept(headers, bodyShallowCopy);
 }
 
-const userFn = loadUserFunction(process.env['SF_FUNCTION_PACKAGE_NAME']);
+
+let userFn;
+try {
+  userFn = loadUserFunction(process.env['SF_FUNCTION_PACKAGE_NAME']);
+} catch(e) {
+  console.error(e);
```

Comment:
use logger instead of console.log?

Reply:
I'll remove the entire try...catch. I don't think we need it. I think it got leftover from a debug session.